src/node_api.cc

```diff
@@ -1409,7 +1409,39 @@ napi_status napi_create_symbol(napi_env env,
   return GET_RETURN_STATUS(env);
 }
 
+static napi_status set_error_code(napi_env env,
+                                  napi_value code,
+                                  const char* code_cstring,
+                                  v8::Local<v8::Value> val) {
```

Comment:
nit: I think this might be clearer if `val` were the second parameter, that’s what I would expect for a setting operation (I’d also call it `obj`, or even `error` – `val` is pretty generic)